pkg/tsdb/cloudwatch/log_query_test.go

```diff
@@ -32,6 +32,10 @@ func TestLogsResultsToDataframes(t *testing.T) {
 					Field: aws.String("line"),
 					Value: aws.String("test message 1"),
 				},
+				&cloudwatchlogs.ResultField{
```

Comment:
This big test that does multiple fields at once is good, but I have found it a bit difficult to run a test which targets the case specifically that this PR targets.

What do you think of a minimal version of this test for this case only like this? 
```
func Test_MixedTypes_NumericValuesMixedWithStringFallBackToStringValues(t *testing.T) {
	dataframes, err := logsResultsToDataframes(&cloudwatchlogs.GetQueryResultsOutput{
		Results: [][]*cloudwatchlogs.ResultField{
			{
				&cloudwatchlogs.ResultField{
					Field: aws.String("numberOrString"),
					Value: aws.String("1"),
				},
			},
			{
				&cloudwatchlogs.ResultField{
					Field: aws.String("numberOrString"),
					Value: aws.String("not a number"),
				},
			},
			{
				&cloudwatchlogs.ResultField{
					Field: aws.String("numberOrString"),
					Value: aws.String("2.000"),
				},
			},
		},
		Status: aws.String("ok"),
	})
	require.NoError(t, err)

	expectedDataframe := &data.Frame{
		Name: "CloudWatchLogsResponse",
		Fields: []*data.Field{
			data.NewField("numberOrString", nil, []*string{
				aws.String("1"),
				aws.String("not a number"),
				aws.String("2.000"),
			}),
		},
		RefID: "",
		Meta: &data.FrameMeta{
			Custom: map[string]interface{}{
				"Status": "ok",
			},
		},
	}

	assert.Equal(t, expectedDataframe.Name, dataframes.Name)
	assert.Equal(t, expectedDataframe.RefID, dataframes.RefID)
	assert.Equal(t, expectedDataframe.Meta, dataframes.Meta)
	assert.ElementsMatch(t, expectedDataframe.Fields, dataframes.Fields)
}
```

My thoughts:
* way easier to read and run individually
* separate test provides clear/concise "documentation" of what you have changed here -- and title can be super specific and read more or less exactly like the the description of the PR
* not a big deal if it is repetitive -- just a test file, not production code. Because of our different goal here, it's ok if this field gets bloated